### ISSUE TITLE
np.searchsorted documentation and typing missing that single values are allowed

### ISSUE THREAD

**Issue by Dr-Irv** [numpy/numpy#19160]
## Documentation

The documentation (and associated typing in the `pyi` files) for `np.searchsorted` is missing that you can pass a single argument. and get a single integer as the return type.

https://numpy.org/doc/stable/reference/generated/numpy.searchsorted.html

Example code that illustrates this:

```python
import numpy as np
a = np.array([1, 3, 5])
# reveal_type(a)
s = a.searchsorted(4)
# reveal_type(s)
print(s, type(s), isinstance(s, np.ndarray))
```

Result of running this is:
```
2 <class 'numpy.int64'> False
```
If I uncomment the `reveal_type` lines, the types of `a` and `s` are both `np.ndarray`.

So the docs should be updated to reflect that you can pass a single value, and, as a result a single integer is returned.  In addition, the typing signature should be fixed as well.

**Comment by BvB93** [comment]
Hi @Dr-Irv, the necessary `ndarray.searchsorted` overload was added back in https://github.com/numpy/numpy/pull/19005, 
so the typing aspect of this issue will be resolved once NumPy 1.21 goes live.
The first release candidate is already up on PyPi, so that's also an option in case you're interested.

**Comment by Dr-Irv** [comment]
@BvB93 That's good news on the typing.  But the docs should be fixed as well.

**Comment by BvB93** [comment]
I agree, yes. A documentation fix is currently up at https://github.com/numpy/numpy/pull/19165.

**Comment by belm0** [comment]
the typing issue can cause mypy errors as a result, e.g.:

```python
index = np.searchsorted(foo, 2)
baz = bar[:index]
```
```
error: Slice index must be an integer or None  [misc]
```

**Comment by BvB93** [comment]
> The typing issue can cause mypy errors as a result. This is a regression from earlier numpy versions.

That would make sense, as numpy versions prior to 1.20 didn't contain any type annotations (_i.e._ `index: Any`).

As for the provided example, I'm afraid that the issue you're looking at is mypy being overly strict (xref https://github.com/python/mypy/issues/2410).
Despite the `slice` class-constructor being annotated as taking any object ([ref](https://github.com/python/typeshed/blob/a4075472d34b59c590047c33f5fc534c925b52c0/stdlib/builtins.pyi#L704-L707)), mypy nevertheless demands 
either `None` or `builtins.int` for the explicit slicing syntax (`:`). 

In contrast, `np.searchsorted`, and many numpy functions in general, return either `np.intp` or an array thereof, neither of which directly inherits from `builtins.int`.
